### DISCUSSION TITLE
Deformable Object Simulation Capability

### DISCUSSION THREAD

**Discussion by austinilo123** [isaac-sim/IsaacLab#587] [Q&A] [answered]
I was wondering when the capability for spawning and interacting with deformable meshes will be added to Isaac Lab. This is currently a feature in Isaac Sim and was shown in the Orbit paper, but the functionality isn't currently implemented in Isaac Lab. 

Is there a timeline on when this will be added?

**Comment by Mayankm96** [comment, marked as answer]
We have started re-working on this: https://github.com/isaac-sim/IsaacLab/pull/613.

The current plan is to merge FEM-based soft body APIs into Isaac Lab and transfer some of the environments (mentioned in the Orbit work) into Isaac Lab. The hope is to have it by the end of this month, but it might get delayed based on other priorities.

The initial Orbit paper used particle-based cloth implementation. Since this was quite difficult to tune for stable simulation, we are waiting on the PhysX team to develop a FEM-based cloth simulation. We will start working on it once it is ready (in the November release of Isaac Sim).

Lastly, the environments are hard to promise for fluid simulation at the moment. Similar to particle-based cloth, the simulation isn't the most stable yet, and the tensor APIs also need to be included. This will probably happen at some point next year.

**Comment by dblanm** [comment]
Hi! I just found this discussion, are there any news on the release date for FEM-based soft body APIs @Mayankm96 ?

**Reply by yu-mei** [reply]
I am also looking forward here!

**Reply by Mayankm96** [reply]
We added the support in release 1.2. Please check the releas notes here: https://github.com/isaac-sim/IsaacLab/releases/tag/v1.2.0

**Reply by dblanm** [reply]
Is there any example for the FEM-based deformable object simulation?

**Reply by Mayankm96** [reply]
Here:
* https://github.com/isaac-sim/IsaacLab/blob/v1.2.0/source/standalone/environments/state_machine/lift_teddy_bear.py
* https://isaac-sim.github.io/IsaacLab/source/overview/showroom.html

Tutorial: https://isaac-sim.github.io/IsaacLab/source/tutorials/01_assets/run_deformable_object.html

**Reply by harshaguda** [reply]
I am also curious about the cloth FEM support, is it already in the pipeline? Can we expect it anytime soon?

**Reply by yu-mei** [reply]
> Here:
> 
> * https://github.com/isaac-sim/IsaacLab/blob/v1.2.0/source/standalone/environments/state_machine/lift_teddy_bear.py
> * https://isaac-sim.github.io/IsaacLab/source/overview/showroom.html
> 
> Tutorial: https://isaac-sim.github.io/IsaacLab/source/tutorials/01_assets/run_deformable_object.html

@Mayankm96 HI Mayankm96, thanks for delivering this exciting version! I am also wondering if the demo of clamped beam is available, which is the case of Fig.11 in the Orbit paper: https://arxiv.org/pdf/2301.04195. Thank you!
